### PR TITLE
Uplink pens now need two separate rotations to unlock

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Uplink pens now require two seperate rotations to unlock. This also
applies to failsafe codes.
/:cl:

With only 359 combinations on an uplink pen, it's crackable given a
small amount of time. With two, it's now 92,981 combinations, which
is more secure than PDA uplink codes (899 * 26 = 23,374).

### CODE REVIEW

**Comment by MACIEKBAKI** [comment]
and this, children, is why you use pens. no one expects it plus they are secure as heck

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -571,3 +571,16 @@
 	for(var/key in input)
 		ret += key
 	return ret
+
+/proc/compare_list(list/l,list/d)
+	if(!istype(l, /list) || !istype(d, /list))
```
Comment (ShizCalev) [inline code/__HELPERS/_lists.dm]: islist

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by kriskog** [comment]
Noone is going to brute force a pen that might not even be the uplink, and if they do they deserve the prize.

**Comment by ExcessiveUseOfCobblestone** [comment]
People have done this before. Pretty sure we had a policy discussion on it.

**Comment by nicbn** [comment]
When will we start hashing and salting uplinks so that database leakages won't expose it?

**Comment by zxaber** [comment]
I dunno, having to input two separate numbers just seems tedious - especially since the uplink closes as soon as you place the pen into your PDA.

**Comment by nicbn** [comment]
if it's that tedious just use pda instead of pen

**Comment by Cyprex** [comment]
You can always drag the PDA on your sprite, pens need to be taken into your hand, correct?

**Comment by zxaber** [comment]
True, but the pens have it worse since there's no special slot, so it's taking up pocket you might want to stick something else in. 

I'm not necessarily saying this is a bad change, I'll just keep on happily using PDA. But I certainly would find two rotations annoying because I wouldn't be able to just easily paste in the code.

**Comment by davethwave** [comment]
You put it in the freaking pda.

**Comment by coiax** [comment]
@ExcessiveUseOfCobblestone y/n?

**Comment by ExcessiveUseOfCobblestone** [comment]
You could just make the code a string of `"[rand(1,360)]-[rand(1,360)]" so you can still copypaste it.

**Comment by Robe21** [comment]
🛌

**Comment by ExcessiveUseOfCobblestone** [comment]
r u going to change it to a string or are you sticking to the two rot method?

**Comment by coiax** [comment]
Changing it to the string method seems... inelegant. They rotate the pen a certain number of degrees, then another. They gain security at the cost of convenience.

**Pull request merged**